### PR TITLE
Fix URLs in CSV exports

### PR DESCRIPTION
**Pull request by thebaer**
This includes the instance's hostname in calls to export a CSV file and `PublicPost.CanonicalURL()`.

It also fixes a panic in that method during CSV export caused by draft posts.

### CODE REVIEW

File: posts.go [hunk 1]
```diff
@@ -1380,12 +1380,14 @@ Are you sure it was ever here?`,
 			IsFound        bool
 			IsAdmin        bool
 			CanInvite      bool
+			Hostname       string
```
Comment (thebaer) [inline posts.go]: The app configuration is passed into every page, so we don't need this extra property. We can just use `.Host` in the templates.

Comment (robjloranger) [reply posts.go]: Oh cool, I can fix it but won't be until later today.

Comment (thebaer) [reply posts.go]: Cool, no worries!

**Comment by robjloranger** [comment]
ok all set

**Comment by thebaer** [comment]
Perfect :+1: Thanks for the review! Merging now.

**Pull request merged**